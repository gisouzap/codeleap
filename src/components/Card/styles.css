.card {
  padding: 24px;
  background-color: var(--color-white);
  border: 1px solid var(--color-grey-200);
  border-radius: 16px;
  width: 100%;
}

.card-input {
  margin: 24px 0px 16px 0px;
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.card-button {
  display: flex;
  justify-content: end;
}

@media (max-width: 480px) {
  .card {
    width: 400px;
  }
}
