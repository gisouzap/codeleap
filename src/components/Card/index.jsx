import PropTypes from 'prop-types';
import './styles.css';

function Card({ children, ...props }) {
  return (
    <div className="card" {...props}>
      {children}
    </div>
  );
}

Card.propTypes = {
  children: PropTypes.node,
};

export default Card;
