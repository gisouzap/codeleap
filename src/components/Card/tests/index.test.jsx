import React from 'react';
import { render } from '@testing-library/react';
import Card from '../index';

describe('Card', () => {
  it('should render card with correct children', () => {
    const { getByText } = render(
      <Card>
        <span>Hello world</span>
      </Card>
    );

    const text = getByText('Hello world');

    expect(text).toBeInTheDocument();
  });

  it('should render and match the snapshot', () => {
    const { container } = render(
      <Card>
        <span>Hello world</span>
      </Card>
    );

    expect(container).toMatchSnapshot();
  });
});
