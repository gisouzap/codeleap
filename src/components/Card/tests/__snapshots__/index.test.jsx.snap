// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render and match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <span>
      Hello world
    </span>
  </div>
</div>
`;
